controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook_test.go

```diff
@@ -744,6 +744,11 @@ func TestKubeadmControlPlaneValidateUpdate(t *testing.T) {
 			before:    before,
 			kcp:       dnsBuildTag,
 		},
+		{
+			name:   "should succeed when using the same CoreDNS version",
```

Comment:
I think we need a test for "should succeed when using the same CoreDNS version - not supported", e.g with CoreDNS v99.99.99